client/js/contextMenu.js

```diff
@@ -0,0 +1,250 @@
+"use strict";
+const $ = require("jquery");
+const socket = require("./socket");
+const utils = require("./utils");
+const templates = require("../views");
+const JoinChannel = require("./join-channel");
+const contextMenuActions = [];
+const contextMenuItems = [];
+let contextMenu, contextMenuContainer;
+
+module.exports = {
+	addContextMenuItem,
+	setUpContextMenu,
+};
+
+/**
+ *
+ * @param opts
```

Comment:
Well, it's jsdoc. Are we really not using it anywhere? It's a standard for documenting functions when you want. I was doing it because the plugins were going to use it, but then I realised they won't get it because of how the plugins work. So no point.

Reply:
I've had poor experience with JSDoc specifically (no need to get into details here, but every time concluded it was not too well cut out for JS, having unnecessary things and missing others), and I've noticed others with similar concerns, which maybe explains why JSDoc is not as widespread as similar standard for other languages.
On the other hand, I've noticed a few project-specific flavors lol.

Anyway, I'm all up for _some kind of doc structure_, because it will be great to extract for the API documentation. Whether we do JSDoc or something else, it doesn't really matter, they all look the same with minor adaptations.

Reply:
I don't know what problems you've had with jsdoc, but it's quite commonly used, and a lot of tooling exists to parse it.

Reply:
I feel like we are focusing on the wrong thing here :-P 

This doc is probably in the wrong place, as the tools won't take this to the API (because users will be getting the API from docs only, their IDEs won't be seeing this code) and parsing it from this function to the docs would be difficult, so *if* we want to keep this (that's a big discussion) it should probably go on the public API bit, not this function.

But this isn't massively important, as the key point is the docs are good.